plugins/net_plugin/net_plugin.cpp

```diff
@@ -3330,22 +3404,31 @@ namespace eosio {
       }
 
       if( accepted ) {
-         boost::asio::post( my_impl->thread_pool.get_executor(), [dispatcher = my_impl->dispatcher.get(), cid=c->connection_id, blk_id, msg]() {
-            fc_dlog( logger, "accepted signed_block : #${n} ${id}...", ("n", msg->block_num())("id", blk_id.str().substr(8,16)) );
-            dispatcher->add_peer_block( blk_id, cid );
+         boost::asio::post( my_impl->thread_pool.get_executor(), [dispatcher = my_impl->dispatcher.get(), c, blk_id, blk_num]() {
+            fc_dlog( logger, "accepted signed_block : #${n} ${id}...", ("n", blk_num)("id", blk_id.str().substr(8,16)) );
+            dispatcher->add_peer_block( blk_id, c->connection_id );
+
+            while (true) { // schedule any possible previously unlinkable blocks
+               unlinkable_block_state prev_unlinkable = my_impl->dispatcher->get_possible_linkable_block(blk_id);
+               if (!prev_unlinkable.block)
+                  break;
+               fc_dlog( logger, "retrying previous unlinkable block #${n} ${id}...",
+                        ("n", block_header::num_from_id(prev_unlinkable.id))("id", prev_unlinkable.id.str().substr(8,16)) );
+               // post at medium_high since this is likely the next block that should be processed (other block processing is at priority::medium)
+               app().executor().post(priority::medium_high, exec_queue::read_write, [prev_unlinkable{std::move(prev_unlinkable)}, c]() mutable {
+                  c->process_signed_block( prev_unlinkable.id, std::move(prev_unlinkable.block), {} );
+               });
+            }
          });
          c->strand.post( [sync_master = my_impl->sync_master.get(), dispatcher = my_impl->dispatcher.get(), c, blk_id, blk_num]() {
             dispatcher->recv_block( c, blk_id, blk_num );
             sync_master->sync_recv_block( c, blk_id, blk_num, true );
          });
       } else {
-         c->strand.post( [sync_master = my_impl->sync_master.get(), dispatcher = my_impl->dispatcher.get(), c, blk_id, blk_num, reason]() {
+         c->strand.post( [sync_master = my_impl->sync_master.get(), dispatcher = my_impl->dispatcher.get(), c,
+                          block{std::move(block)}, blk_id, blk_num, reason]() mutable {
             if( reason == unlinkable || reason == no_reason ) {
-               // unlinkable may be linkable in the future, so indicate we have not received it
-               // call on dispatch strand to serialize with the add_peer_block calls
-               my_impl->dispatcher->strand.post( [blk_id]() {
-                  my_impl->dispatcher->rm_block( blk_id );
-               } );
+               my_impl->dispatcher->add_unlinkable_block( std::move(block), blk_id );
```

Comment:
Can `my_impl->dispatcher` jus be `dispatcher`?

Reply:
Yes